### PR TITLE
Detect flutter or flutter_test deps and fail eagerly

### PR DESCRIPTION
**Pull request by jakemac53**
Towards https://github.com/dart-lang/webdev/issues/359

### CODE REVIEW

**Comment by kevmoo** [comment]
cupertino_icons
shared_preferences
url_launcher
path_provider
firebase_auth
image_picker
cloud_firestore
sqflite
cached_network_image
google_sign_in
firebase_core

These are the most popular flutter-requiring plugins.

**Comment by jakemac53** [comment]
Added the extra packages - but I only left flutter/flutter_test in for the test as the additional ones don't really provide much value in that context.

File: webdev/lib/src/pubspec.dart [hunk 1]
```diff
@@ -48,6 +48,12 @@ dev_dependencies:
   $pkgName: $constraint''',
           missingDependency: true);
 
+  static PackageExceptionDetails unsupportedDep(String pkgName) =>
```
Comment (kevmoo) [inline webdev/lib/src/pubspec.dart]: rename this method to `unsupportedFlutterWebDep` or similar?

File: webdev/lib/src/pubspec.dart [hunk 2]
```diff
@@ -178,6 +193,26 @@ Future<void> checkPubspecLock(PubspecLock pubspecLock,
     issues.addAll(await _validateBuildDaemonVersion(pubspecLock));
   }
 
+  var unsupportedFlutterDeps = [
```
Comment (natebosch) [inline webdev/lib/src/pubspec.dart]: I know I'm late - but I really don't like the idea of hardcoding a bunch of packages like this.

We don't hardcoded knowledge that `package:isolate` will only ever be VM only and search for it...
Are we planning on expanding if other SDKs start popping up?

What if we instead search the pubspec.lock and find packages with type "sdk". We know that the web SDK doesn't ship with packages, and so seeing those would be a more general solution.

**Pull request merged**